src/channels.c

```diff
@@ -665,6 +662,22 @@ channel_get_icon ( channel_t *ch )
           s++;
         }
       }
+      else if((send = strstr(chi, "%c"))) {
+        char *aname = intlconv_utf8safestr(intlconv_charset_id("ASCII", 1, 1),
+                                     chname, strlen(chname) * 2);
+        sname = url_encode(aname);
```

Comment:
Missing NULL check for aname. The NULL can be returned if iconv() function fails for a reason.